tests/cpp/test_persistent_buffer.cpp

```diff
@@ -1456,4 +1457,121 @@ TEST_F(PersistentBufferTest, InnerPersistentNotEnoughSharedMemory) {
+  SegmentCandidateFinder::translateWelfordInFusion(&fusion, runtime_inputs);
+  auto fusion_copy = fusion;
+
+  // check persistent buffer size
+  SchedulerRuntimeInfo runtime_info(
+      &fusion, {aten_input, aten_weight, aten_bias});
+  auto persistent_buffer_info = scheduler_utils::persistentBuffers(&fusion);
+  auto persistent_buffer_size =
+      persistentBufferSize(&fusion, runtime_info, persistent_buffer_info);
+  int64_t logic_buffer_size = hidden_size * dataTypeSize(dtype);
+  EXPECT_EQ(
+      persistent_buffer_size.projected_persistent_buffer_size,
+      logic_buffer_size);
+
+  // expect segmentation?
+  bool has_enough_regs_smem = true;
+  if (logic_buffer_size > scheduler_utils::register_file_size) {
+    const auto dev_prop = at::cuda::getCurrentDeviceProperties();
+    if ((int64_t)dev_prop->sharedMemPerMultiprocessor < logic_buffer_size) {
+      has_enough_regs_smem = false;
+    } else {
+      int64_t available_buffer_size = normalization_scheduler_utils::
+          getMaxRegOrSharedMemorySizeForPersistentBuffer(
+              &fusion,
+              runtime_info,
+              scheduler_utils::getReductionTvs(&fusion),
+              persistent_buffer_info,
+              /*can_use_smem_persistent*/ true,
+              /*project_to_inputs*/ true);
+      has_enough_regs_smem = available_buffer_size >= logic_buffer_size;
+    }
+  }
+
+  // check segmentation and smem usage
+  FusionExecutorCache executor_cache(std::move(fusion_ptr));
+  auto cg_outputs = executor_cache.runFusionWithInputs(aten_inputs);
+  auto runtime = executor_cache.getMostRecentKernelRuntime();
+  if (has_enough_regs_smem) {
+    // For dtype float, no op scheduler is also used.
+    const auto& heuristic_param =
+        runtime->schedulerHeuristics()->heuristicsList().back();
+    EXPECT_EQ(heuristic_param->scheduler_type, SchedulerType::InnerPersistent);
+    Fusion* scheduled_fusion = runtime->executors()
+                                   .back()
+                                   ->as<KernelExecutor>()
+                                   ->compiledKernel()
+                                   ->kernel();
+
+    if (logic_buffer_size > scheduler_utils::register_file_size) {
+      bool has_smem_tv = false;
+      for (auto tv : scheduled_fusion->allTvs()) {
+        if (tv->getMemoryType() == MemoryType::Shared) {
+          has_smem_tv = true;
+          break;
+        }
+      }
+      EXPECT_TRUE(has_smem_tv);
+    }
+  } else {
+    EXPECT_THAT(
+        runtime->fusionSegments()->groups(),
+        Contains(HeuristicIs(SchedulerType::Reduction)));
+  }
+  testValidate(&fusion_copy, cg_outputs, aten_inputs, __LINE__, __FILE__, "");
+}
+INSTANTIATE_TEST_SUITE_P(
+    PersistentBufferTest,
+    LayerNormSharedMemoryTest,
+    ::testing::Combine(
+        ::testing::Values(DataType::Half, DataType::Float),
+        ::testing::Range((int64_t)32768, (int64_t)81921, (int64_t)4096)),
```

Comment:
so the old hidden size goes from (8192, 131072, 8192)?

Is this an intended change?

Reply:
yes, reduced step size for better coverage and decrease range to reduce cost. Shared memory persistence starts from 32K for float and 64K for float16, no need to test very small or very large inputs.